lib/matplotlib/figure.py

```diff
@@ -1262,127 +1262,213 @@ def draw_artist(self, a):
+
+        labelspacing : float or None
+            The vertical space between the legend entries, measured in
+            font-size units.
+            Default is *None* which will take the value from the
+            ``legend.labelspacing`` :data:`rcParam<matplotlib.rcParams>`.
+
+        handlelength : float or None
+            The length of the legend handles, measured in font-size units.
+            Default is *None* which will take the value from the
+            ``legend.handlelength`` :data:`rcParam<matplotlib.rcParams>`.
+
+        handletextpad : float or None
+            The padding between the legend handle and text, measured in
+            font-size units.
+            Default is *None* which will take the value from the
+            ``legend.handletextpad`` :data:`rcParam<matplotlib.rcParams>`.
 
-        Padding and spacing between various elements use following keywords
-        parameters. The dimensions of these values are given as a fraction
-        of the fontsize. Values from rcParams will be used if None.
+        borderaxespad : float or None
+            The padding between the axes and legend border, measured in
+            font-size units.
+            Default is *None* which will take the value from the
+            ``legend.borderaxespad`` :data:`rcParam<matplotlib.rcParams>`.
 
-        ================   ====================================================
-        Keyword            Description
-        ================   ====================================================
-        borderpad          the fractional whitespace inside the legend border
-        labelspacing       the vertical space between the legend entries
-        handlelength       the length of the legend handles
-        handletextpad      the pad between the legend handle and text
-        borderaxespad      the pad between the axes and legend border
-        columnspacing      the spacing between columns
-        ================   ====================================================
+        columnspacing : float or None
+            The spacing between columns, measured in font-size units.
+            Default is *None* which will take the value from the
+            ``legend.columnspacing`` :data:`rcParam<matplotlib.rcParams>`.
 
-        .. Note:: Not all kinds of artist are supported by the legend.
-                  See LINK (FIXME) for details.
+        Returns
+        -------
+        :class:`matplotlib.legend.Legend` instance
 
-        **Example:**
+        Notes
+        -----
+        Not all kinds of artist are supported by the legend command.
+        See :ref:`plotting-guide-legend` for details.
 
+        Examples
+        --------
         .. plot:: mpl_examples/pylab_examples/figlegend_demo.py
         """
-        l = Legend(self, handles, labels, *args, **kwargs)
+
+        if len(args) == 0:
+            handles = []
+            labels = []
+
+            def in_handles(h, l):
```

Comment:
Sounds good.  Typo: "consdier".
I would make it a comment rather than a docstring to save some whitespace, and get rid of the whitespace between the try... catch blocks (right now it's very spread out).